serde/src/lib.rs

```diff
@@ -10,36 +10,64 @@ pub use serde_derive::{Deserialize, Serialize};
 
 /// Deserialization Error. This error is a wrapper over `postcard::Error`.
 #[derive(Eq, PartialEq)]
-pub struct DeserializeError(postcard::Error);
+pub struct DeserializeError(DeserializeErrorInner);
+
+#[derive(Eq, PartialEq)]
+enum DeserializeErrorInner {
```

Comment:
Isn't it better to derive the error using `thiserror`?

Reply:
I think I prefer the current way because it minimizes the length of the `Debug` implementation, since it directly reduces to the error it's wrapping.